### ISSUE TITLE
Importing new recipes fails due to unique tags

### ISSUE THREAD

**Issue by hamperfait** [mealie-recipes/mealie#699]
**Describe the bug**

I have been trying to import recipes via zipfiles with new tags to mealie and not been successful at all. Whenever I upload the recipe with tags I get an error, and the docker logs state:
```
mealie      | DETAIL:  Key (slug)=(party) already exists.
mealie      | 
mealie      | [SQL: INSERT INTO tags (name, slug) VALUES (%(name)s, %(slug)s) RETURNING tags.id]
```

I'm guessing this happens because the tags have been set to unique somewhere and it does not allow to associate recipes with those tags, but that might not be right because I have just started to use mealie, and so far I don't have any tags at all, so I don't know what it could be.

I have tried using both MySQL and PostgreSQL, both gave the same error.

*TL;DR:*

When I import a recipe in a zipfile that contains tags, upload fails because the tag already exists and recipes are not imported

**Device Information (please complete the following information):**
 - OS: Debian
 - Deployment: docker-compose
 - Browser: chrome
 - Version: 3.1

Here is the full stacktrace
```
mealie      | 172.20.0.1:0 - "POST /api/recipes/create-from-zip HTTP/1.1" 500
mealie      | [2021-09-28 10:43:51 -0800] [48] [ERROR] Exception in ASGI application
mealie      | Traceback (most recent call last):
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi
mealie      |     result = await app(self.scope, self.receive, self.send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
mealie      |     return await self.app(scope, receive, send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/fastapi/applications.py", line 199, in __call__
mealie      |     await super().__call__(scope, receive, send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/applications.py", line 111, in __call__
mealie      |     await self.middleware_stack(scope, receive, send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
mealie      |     raise exc from None
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
mealie      |     await self.app(scope, receive, _send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
mealie      |     raise exc from None
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
mealie      |     await self.app(scope, receive, sender)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/routing.py", line 566, in __call__
mealie      |     await route.handle(scope, receive, send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/routing.py", line 227, in handle
mealie      |     await self.app(scope, receive, send)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/starlette/routing.py", line 41, in app
mealie      |     response = await func(request)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/fastapi/routing.py", line 201, in app
mealie      |     raw_response = await run_endpoint_function(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/fastapi/routing.py", line 148, in run_endpoint_function
mealie      |     return await dependant.call(**values)
mealie      |   File "/app/mealie/routes/recipe/recipe_crud_routes.py", line 119, in create_recipe_from_zip
mealie      |     recipe: Recipe = db.recipes.create(session, Recipe(**recipe_dict))
mealie      |   File "/app/mealie/db/db_base.py", line 130, in create
mealie      |     session.commit()
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 1428, in commit
mealie      |     self._transaction.commit(_to_root=self.future)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 829, in commit
mealie      |     self._prepare_impl()
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 808, in _prepare_impl
mealie      |     self.session.flush()
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3298, in flush
mealie      |     self._flush(objects)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3438, in _flush
mealie      |     transaction.rollback(_capture_exception=True)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
mealie      |     compat.raise_(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
mealie      |     raise exception
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/session.py", line 3398, in _flush
mealie      |     flush_context.execute()
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 456, in execute
mealie      |     rec.execute(self)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/unitofwork.py", line 630, in execute
mealie      |     util.preloaded.orm_persistence.save_obj(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 242, in save_obj
mealie      |     _emit_insert_statements(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/orm/persistence.py", line 1153, in _emit_insert_statements
mealie      |     c = connection._execute_20(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1583, in _execute_20
mealie      |     return meth(self, args_10style, kwargs_10style, execution_options)
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
mealie      |     return connection._execute_clauseelement(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1452, in _execute_clauseelement
mealie      |     ret = self._execute_context(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1814, in _execute_context
mealie      |     self._handle_dbapi_exception(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1995, in _handle_dbapi_exception
mealie      |     util.raise_(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/util/compat.py", line 207, in raise_
mealie      |     raise exception
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/engine/base.py", line 1751, in _execute_context
mealie      |     self.dialect.do_executemany(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 962, in do_executemany
mealie      |     context._psycopg2_fetched_rows = xtras.execute_values(
mealie      |   File "/opt/pysetup/.venv/lib/python3.9/site-packages/psycopg2/extras.py", line 1270, in execute_values
mealie      |     cur.execute(b''.join(parts))
mealie      | sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "ix_tags_slug"
mealie      | DETAIL:  Key (slug)=(party) already exists.
mealie      | 
mealie      | [SQL: INSERT INTO tags (name, slug) VALUES (%(name)s, %(slug)s) RETURNING tags.id]
mealie      | [parameters: ({'name': 'Afternoon tea', 'slug': 'afternoon-tea'}, {'name': 'Snack', 'slug': 'snack'}, {'name': 'Dessert', 'slug': 'dessert'}, {'name': 'Bread', 'slug': 'bread'}, {'name': 'Baking', 'slug': 'baking'}, {'name': 'Contains egg', 'slug': 'contains-egg'}, {'name': 'Contains gluten', 'slug': 'contains-gluten'}, {'name': 'Contains lactose', 'slug': 'contains-lactose'}  ... displaying 10 of 28 total bound parameter sets ...  {'name': 'Party', 'slug': 'party'}, {'name': 'Vegetarian', 'slug': 'vegetarian'})]
mealie      | (Background on this error at: https://sqlalche.me/e/14/gkpj)

```

**Comment by cadamswaite** [comment]
I'm struggling to recreate this. Please could you see if you still see the issue in the latest mealie release (0.5.3)?
If you do, please can you provide an example zip file that fails

**Comment by hamperfait** [comment]
Hi, I realized the issue was that my recipe had many tags, and two were the same tag, so that's why it failed. I'm closing the issue.
